spatialdata/_core/_spatialdata.py

```diff
@@ -48,163 +50,301 @@ class SpatialData:
+                if name in elem_group:
+                    raise ValueError(f"Element {name} already exists, use overwrite=True to overwrite it")
+
+        if element_type != "labels":
+            return elem_group
+        else:
+            return root
+
+    def add_image(
+        self,
+        name: str,
+        image: Union[SpatialImage, MultiscaleSpatialImage],
+        storage_options: Optional[Union[JSONDict, List[JSONDict]]] = None,
+        overwrite: bool = False,
+    ) -> None:
+        # _init_add_element() needs to be called before _add_image_in_memory(), and same for the other elements
+        # otherwise if a element is added and saved to disk, and another element with the same name but different
+        # content is added, _init_add_element() will raise an exception. If _add_image_in_memory() is called first,
+        # then the memory content will be overwritten, but the disk content will remain the old one
+        elem_group = self._init_add_element(name=name, element_type="images", overwrite=overwrite)
+        if name not in self.images:
+            self._add_image_in_memory(name=name, image=image)
+        write_image(
+            image=self.images[name],
+            group=elem_group,
+            name=name,
+            storage_options=storage_options,
+        )
+
+    def add_labels(
         self,
-        file_path: str,
+        name: str,
+        labels: Union[SpatialImage, MultiscaleSpatialImage],
         storage_options: Optional[Union[JSONDict, List[JSONDict]]] = None,
+        overwrite: bool = False,
     ) -> None:
-        """Write to Zarr file."""
+        elem_group = self._init_add_element(name=name, element_type="labels", overwrite=overwrite)
+        if name not in self.labels:
+            self._add_labels_in_memory(name=name, labels=labels)
+        write_labels(
+            labels=self.labels[name],
+            group=elem_group,
+            name=name,
+            storage_options=storage_options,
+        )
+
+    def add_points(
+        self,
+        name: str,
+        points: pa.Table,
+        overwrite: bool = False,
+    ) -> None:
+        elem_group = self._init_add_element(name=name, element_type="points", overwrite=overwrite)
+        if name not in self.points:
+            self._add_points_in_memory(name=name, points=points)
+        write_points(
+            points=self.points[name],
+            group=elem_group,
+            name=name,
+        )
+
+    def add_polygons(
+        self,
+        name: str,
+        polygons: GeoDataFrame,
+        overwrite: bool = False,
+    ) -> None:
+        elem_group = self._init_add_element(name=name, element_type="polygons", overwrite=overwrite)
+        if name not in self.polygons:
+            self._add_polygons_in_memory(name=name, polygons=polygons)
+        write_polygons(
+            polygons=self.polygons[name],
+            group=elem_group,
+            name=name,
+        )
+
+    def add_shapes(
```

Comment:
please add a doctsring. at a minimum, please add a description of what the function does.

Reply:
done